src/NServiceBus.Core/Recoverability/RecoverabilityContext.cs

```diff
@@ -4,26 +4,41 @@
     using System.Collections;
     using System.Collections.Generic;
     using System.Linq;
+    using NServiceBus.Transport;
     using Pipeline;
-    using Transport;
 
     class RecoverabilityContext : BehaviorContext, IRecoverabilityContext, IRecoverabilityActionContext, IRecoverabilityActionContextNotifications
     {
         public RecoverabilityContext(
-            ErrorContext errorContext,
```

Comment:
I would have preferred a design where we still pass the ErrorContext as a parameter because it would make it clear this context can only be instantiated where the error context is available. If you prefer this approach I'm also ok to go with this one

Reply:
Since the ctor isn't exposed I think this is a good idea, will change

Reply:
done